bpf/bpf.go

```diff
@@ -360,6 +437,54 @@ func (b *BPFKProbePerf) readVersion() (int, error) {
 	return 0, nil
 }
 
+func utsnameStr(in []int8) string {
+	out := make([]byte, len(in))
+
+	for i := 0; i < len(in); i++ {
+		if in[i] == 0 {
+			break
+		}
+		out = append(out, byte(in[i]))
+	}
+
+	return string(out)
+}
+
+func currentVersion() (int, error) {
+	var buf syscall.Utsname
+	if err := syscall.Uname(&buf); err != nil {
+		return -1, err
+	}
+
+	releaseStr := strings.Trim(utsnameStr(buf.Release[:]), "\x00")
+
+	kernelVersionStr := strings.Split(releaseStr, "-")[0]
```

Comment:
Doesn't break on my Debian box but the result doesn't reflect the current kernel version it seems: uname -r is 4.8.0-2-amd64 (kernel version 264192) whereas the actual version seems to be 4.8.11 (264203) and the prog cannot be loaded..